src/lib/statistics/chunk_statistics/abstract_filter.hpp

```diff
@@ -17,14 +17,15 @@ class AbstractFilter : public std::enable_shared_from_this<AbstractFilter> {
   virtual ~AbstractFilter() = default;
 
   /**
-   * checks whether the filter is able to determine that the given value 
+   * checks whether the filter is able to determine that the given value
    * and predicate condition will not yield any positive results with the values
    * represented by the filter data.
-   * 
+   *
    * In other words: returns true if a scan operation with value and predicate_type
    * on the segment that this filter was created on would yield zero result rows.
   */
-  virtual bool can_prune(const AllTypeVariant& value, const PredicateCondition predicate_type) const = 0;
+  virtual bool can_prune(const PredicateCondition predicate_type, const AllTypeVariant& variant_value,
```

Comment:
Yep, I agree, we talked about this already.
I'd hate to take out the functionality of BETWEEN, because it is better than estimating twice with <= and >=.